app/Http/Controllers/PrintController.php

```diff
@@ -14,27 +14,23 @@ public function index() {
     }
 
     public function modify_balance(Request $request) {
-        $balance = $request->balance;
-        $print_account = User::findOrFail($request->user_id)->printAccount; 
         $validator = Validator::make($request->all(), [
             'user_id' => 'required|integer|exists:users,id',
             'balance' => 'required|integer',
         ]);
         $validator->validate();
 
-        if ($balance < 0) {
-            $balance = abs($balance);
-            if ($print_account->balance < $balance) {
-                $validator->errors()->add('balance', __('print.nobalance'));
-                return back()
-                        ->withErrors($validator)
-                        ->withInput();
-            } else {
-                $print_account->decrement('balance', $balance);
-            }
-        } else {
-            $print_account->increment('balance', $balance);
+        $balance = $request->balance;
+        $print_account = User::find($request->user_id)->printAccount;
```

Comment:
Please use User::findOrFail

Reply:
Does not matter here. The user ID will be valid, since the validator checked it for us.
Even if the find would fail, we would use the result of that search, so would get an exception in that case.

Did I miss something?

Reply:
Up to you, it's the safer choice. There is an unlikely scenario when the user disappears from the db between the validation and the call. The generated error would be easier to debug.

But that case is so unlikely, that it can be ignored.